### PR TITLE
build: update `canonical-path` dependency

### PR DESCRIPTION
**Pull request by petebacondarwin**
This new version (1.0.0) provides a typings file!

### CODE REVIEW

**Review by alexeagle — APPROVED** [review]
I like how you've fixed the tools/npm/yarn.lock to work with currently mandated version of yarn...

**Comment by petebacondarwin** [comment]
@alexeagle - are you able to do the g3sync for this?

**Comment by petebacondarwin** [comment]
Caretaker I believe that this PR does not need a g3 sync (ask @alexeagle).

File: packages/compiler-cli/src/ngcc/test/BUILD.bazel [hunk 1]
```diff
@@ -15,6 +15,7 @@ ts_library(
         "//packages/compiler-cli/src/ngtsc/transform",
         "@ngdeps//@types/convert-source-map",
         "@ngdeps//@types/mock-fs",
+        "@ngdeps//canonical-path",
```
Comment (IgorMinar) [inline packages/compiler-cli/src/ngcc/test/BUILD.bazel]: isn't it odd that we needed to add this dep now? why now?

**Review by IgorMinar — APPROVED** [review]
one Q. otherwise lgtm

**Comment by petebacondarwin** [comment]
I believe `test_ivy_aot` error is inherited from master.

**Pull request closed without merging**